structuralcodes/sections/specific.py

```diff
@@ -0,0 +1,446 @@
+    along local x and y axes.
+
+    Input parameters:
+    width: width of the section
+    height: height of the section
+    cover: net cover (distance from outer edget of the section to
+    edge of the stirrup)
+    reinforcement_data: ReinforcementData
+    concrete: Concrete
+    reinforcement: Reinforcement (TODO!!!!!! for now const law)
+    confined_concrete: Concrete (default = None) When set to None the
+    same concrete is adopted for concrete core. When using the string
+    "auto" the confinement will be automatically computed
+    stirrups_data: StirrupsData (default = None) to use when need
+    automatic computation of concrete confinement
+    name: optional string for identidying the section
+
+    Axes definition:
+
+                     z ▲
+                       │
+                       │
+            ┌──────────┼───────────┐
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          │           │
+            │          └───────────┼─────►
+            │                      │     y
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            │                      │
+            └──────────────────────┘
+
+            RectangularRCSection.fromGeometry(Geometry(itPolygons,itMaterials))
+
+
+    """
+
+    def __init__(
+        self,
+        width: float,
+        height: float,
+        cover: float,
+        reinforcement_data: ReinforcementData,
+        concrete: Concrete,
+        reinforcement: ElasticPlastic,
+        confined_concrete: t.Union[Concrete, str, None] = None,
+        stirrups_data: t.Optional[StirrupsData] = None,
+        name: t.Optional[str] = None,
+    ) -> None:
+        if name is None:
+            name = 'RectangularRCSection'
+        super().__init__(name)
+
+        # section geometry
+        self.height = height
+        self.width = width
+        self.cover = cover
+
+        # materials
+        self.concrete = concrete
+        self.reinforcement = reinforcement
+        if confined_concrete is not None:
+            if isinstance(confined_concrete, str):
+                # provided a string, check if it auto
+                if (
+                    confined_concrete.lower() == 'automatic'
+                    or confined_concrete.lower() == 'auto'
+                ):
+                    # TODO: compute automatic confinement
+                    if stirrups_data is None:
+                        raise ValueError(
+                            'Asked for automatic confinement \
+                            computation, but stirrups_data is None \
+                            \nPlease provide a proper stirrups_data'
+                        )
```

Comment:
These lines and lines +179 to +182 are somehow incompatible: if `None` is provided we assume a default 182. We should behave coherently here and there.

Reply:
Not relevant (this is your old code which I have pasted in as a help)